### PR TITLE
BST-3366 Adds CWE mapping to semgrep.

### PR DESCRIPTION
**Pull request by stlef14**
The CWE missing and impacting Coupa:
CWE-328 and CWE-1333.
this adds CWE-328

### CODE REVIEW

File: scanners/boostsecurityio/semgrep/rules.yaml [hunk 1]
```diff
@@ -241,6 +241,19 @@ rules:
     description: The use of a broken or risky cryptographic algorithm is an unnecessary
       risk that may result in the exposure of sensitive information.
     ref: https://github.com/returntocorp/semgrep-rules/
+  CWE-328:
+    categories:
+    - ALL
+    - cwe-328
```
Comment (SUSTAPLE117) [inline scanners/boostsecurityio/semgrep/rules.yaml:247]: @fproulx-boostsecurity @Becojo @stlef14 do we have that cwe category in the database?

Comment (stlef14) [reply scanners/boostsecurityio/semgrep/rules.yaml:247]: @SUSTAPLE117 yes, I checked. 
    {
        "name": "cwe-328",
        "pretty_name": "CWE-328: Use of Weak Hash",
        "doc_ref": "https://cwe.mitre.org/data/definitions/328.html",
    },

Comment (stlef14) [reply scanners/boostsecurityio/semgrep/rules.yaml:247]: that's in the rules-management alembic revision

Comment (SUSTAPLE117) [reply scanners/boostsecurityio/semgrep/rules.yaml:247]: @stlef14 perfect!

**Review by SUSTAPLE117 — APPROVED** [review]

**Pull request merged**